services/121-service/src/payments/fsp-integration/intersolve-visa/intersolve-visa.api.service.ts

```diff
@@ -616,33 +617,41 @@ export class IntersolveVisaApiService {
     return;
   }
 
-  // TODO: This function should throw an expection if the response contains errors, like the other (re-implemented) functions do.
-  public async toggleBlockWallet({
-    tokenCode,
-    payload,
-    block,
-  }: {
-    tokenCode: string | null;
-    payload: IntersolveBlockWalletDto;
-    block: boolean;
-  }): Promise<IntersolveBlockWalletResponseDto> {
+  public async setTokenBlocked(
+    tokenCode: string,
+    blocked: boolean,
+  ): Promise<IntersolveApiBlockWalletResponseDto> {
     const authToken = await this.getAuthenticationToken();
     const apiPath = process.env.INTERSOLVE_VISA_PROD
       ? 'pointofsale-payments'
       : 'pointofsale';
     const url = `${intersolveVisaApiUrl}/${apiPath}/v1/tokens/${tokenCode}/${
-      block ? 'block' : 'unblock'
+      blocked ? 'block' : 'unblock'
     }`;
     const headers = [
       { name: 'Authorization', value: `Bearer ${authToken}` },
       { name: 'Tenant-ID', value: process.env.INTERSOLVE_VISA_TENANT_ID },
     ];
+    const payload = {
+      reasonCode: blocked
+        ? BlockReasonEnum.BLOCK_GENERAL
```

Comment:
I think we can put a refactor comment here to take out this complex DTO - enum structure. It is used only here. Except for short-term also in the migrate code. So after that is done, it can be refactored :)

Reply:
I'll add typechecking for reasonCode and remove/simplify the enum